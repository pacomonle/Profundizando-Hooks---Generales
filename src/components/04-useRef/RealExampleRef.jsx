import React, { useState } from 'react'
import MultipleCustomsHooks from '../03-examples/MultipleCustomsHooks';
import '../02-useEffect/effects.css'

const RealExampleRef = () => {


const [show, setshow] = useState(false)



    return (
        <div>
            <h1>Hook useRef</h1>
            <h3>Ejemplo real</h3>
            <hr />
            {
                show &&  <MultipleCustomsHooks></MultipleCustomsHooks>
            }
            <button 
               type="button" 
               onClick = {() => setshow(!show)}
               className='btn btn-outline-danger btn-block my-5'>
               {!show ? 'Mostrar' : 'Ocultar'}
            </button>
        </div>
    )
}

export default RealExampleRef
