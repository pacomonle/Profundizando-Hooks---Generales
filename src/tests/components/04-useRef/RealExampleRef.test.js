import React from 'react';
import { shallow } from 'enzyme';
import  RealExampleRef  from '../../../components/04-useRef/RealExampleRef';
import  MultipleCustomHooks  from '../../../components/03-examples/MultipleCustomsHooks';

describe('Pruebas en <RealExampleRef />', () => {
    
   // const [show, setshow] = useState(false) -> <RealExampleRef />

    const wrapper = shallow( <RealExampleRef /> );
    test('debe mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find(MultipleCustomHooks).exists() ).toBe(false);

    });

    test('debe de mostrar el componente <MultipleCustomHooks />', () => {
    
        wrapper.find('button').simulate('click');
        expect( wrapper.find(MultipleCustomHooks).exists() ).toBe(true);
    })
    

  
})
