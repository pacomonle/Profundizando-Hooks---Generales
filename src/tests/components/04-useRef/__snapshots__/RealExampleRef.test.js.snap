// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Hook useRef
  </h1>
  <h3>
    Ejemplo real
  </h3>
  <hr />
  <button
    className="btn btn-outline-danger btn-block my-5"
    onClick={[Function]}
    type="button"
  >
    Mostrar
  </button>
</div>
`;
